layer4/evaluation_plan.go

```diff
@@ -0,0 +1,165 @@
+		controlName := "Unnamed Control"
+		if plan.Control.EntryId != "" {
+			controlName = plan.Control.EntryId
+		} else if plan.Control.ReferenceId != "" {
+			controlName = plan.Control.ReferenceId
+		}
+
+		// Format control reference as "Framework / Control-ID" (e.g. OSPS-B / OSPS-AC-01)
+		controlReference := ""
+		if plan.Control.ReferenceId != "" || plan.Control.EntryId != "" {
+			controlReference = fmt.Sprintf("%s / %s", plan.Control.ReferenceId, plan.Control.EntryId)
+		}
+
+		items, err := buildChecklistItems(&plan)
+		if err != nil {
+			return Checklist{}, fmt.Errorf("failed to build checklist items for control %q: %w", controlName, err)
+		}
+
+		section := ControlSection{
+			ControlName:      controlName,
+			ControlReference: controlReference,
+			Items:            items,
+		}
+
+		checklist.Sections = append(checklist.Sections, section)
+	}
+
+	return checklist, nil
+}
+
+// ToMarkdownChecklist converts an evaluation plan into a markdown checklist.
+// Generates a pre-execution checklist showing what needs to be checked.
+func (e EvaluationPlan) ToMarkdownChecklist() (string, error) {
+	checklist, err := e.ToChecklist()
+	if err != nil {
+		return "", fmt.Errorf("failed to build checklist: %w", err)
+	}
+
+	tmpl, err := template.New("checklist").Parse(MarkdownTemplate)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse template: %w", err)
+	}
+
+	var buf bytes.Buffer
+	if err := tmpl.Execute(&buf, checklist); err != nil {
+		return "", fmt.Errorf("failed to execute template: %w", err)
+	}
+
+	return buf.String(), nil
+}
+
+// buildChecklistItems converts an AssessmentPlan into checklist items.
+func buildChecklistItems(plan *AssessmentPlan) ([]ChecklistItem, error) {
+	if plan == nil {
+		return nil, fmt.Errorf("assessment plan is nil")
+	}
+
+	if len(plan.Assessments) == 0 {
+		return nil, fmt.Errorf("assessment plan has no assessments")
+	}
+
+	var items []ChecklistItem
+	assessmentNum := 1
+
+	for _, assessment := range plan.Assessments {
+		requirementId := assessment.Requirement.EntryId
+		if requirementId == "" {
+			requirementId = fmt.Sprintf("Assessment %d", assessmentNum)
+		}
+		assessmentNum++
+
+		if len(assessment.Procedures) == 0 {
+			return nil, fmt.Errorf("assessment %q has no procedures", requirementId)
+		}
+
+		for i, procedure := range assessment.Procedures {
+			// Get procedure name with fallback: Name -> Description -> Id
+			procedureName := procedure.Id
+			if procedure.Name != "" {
+				procedureName = procedure.Name
+			} else if procedure.Description != "" {
+				procedureName = procedure.Description
+			}
+
+			item := ChecklistItem{
+				RequirementId:         requirementId,
+				ProcedureName:         procedureName,
+				Description:           procedure.Description,
+				Documentation:         procedure.Documentation,
+				IsAdditionalProcedure: i > 0,
+			}
+
+			// RequirementId cleared for subsequent procedures and visual hierarchy
```

Comment:
I think it would better if we left visual considerations in the template and reduce data loss in the extraction logic. WDYT?